### DISCUSSION TITLE
is it possible to track statistics per lap (custom startpoint)

### DISCUSSION THREAD

**Discussion by c33s** [OpenTracksApp/OpenTracks#1555] [Q&A]
in my example image the green circle represents the lap which should be tracked. the blue line represents the "travel to location" and the pink line the "travel from location". 
![image](https://user-images.githubusercontent.com/649209/236624452-5a4868fb-7cc3-4b04-ac10-3fae2c630633.png)

i would love to track the whole time without splitting it into multiple "records" (means the to location, the from location part and multiple laps i will take there). i think this would need a kind of button to let opentracks know that a specific point is the start and endpoint for counting laps.

is there such a lap feature? the closest feature i found would be the intervals and pace but it would require at least custom intervals and a custom pace distance.

thank you for this awesome app (please readd heart track history soon, it is a real bummer that the refactoring removed that feature)

**Comment by dennisguse** [comment]
I guess this functionality may already be there: you can just stop and restart the recording.
For OpenTracks this splits the track into separate segments (in the code `SEGMENT_START_MANUAL`).

On export this is also included in KML (first point in a segment has no location or speed).
In GPX this cannot be distinguished as all points require a location.

Then you just have to explain this to your analysis software :)

**Comment by dennisguse** [comment]
And we have no lap feature (based upon reaching the same location multiple times).

Do you need the statistics during the exercise or afterwards?
And which tool do you use to do your analytics?

**Comment by dennisguse** [comment]
After sleeping one night over it: would it make more sense to integrate something like this into Fitrackee?

My argument goes along the line that OprnTracks is running on a mobile only and fancy statistics are not that easy to do there; especially due to the usually limited screen size. 
And this functionality might also be interesting to a lot of non-OpenTracks users ;)

@SamR1 FYI

**Comment by c33s** [comment]
> you can just stop and restart the recording.

yeah that could be a solution.


i know the following is more a dream than reality which will come to the app but i would like to share it anyway. it would be awsome to have the following features directly baked in the app:

- mark a single point as start/stop point and make the app automatically track the laps when passing the geo coordinates again. tracking and displaying the following "
    - laps: 4
    - best lap: 2
    - best lap time: 3:21
    - average lap time: 4:02
    - last lap: 3:31
- mark multiple lap segments start and endpoints, maybe even add a workout type for them (biking, running, walking, swimming, ...). so you can do your laps (where one lap consists out of multilple segments (start walking for the first 100m, then run 500m, then swim 100m, comming back to the start point where you start walking again)

if you really think about some of the features, i can describe them more detailed and add some mokups

> Then you just have to explain this to your analysis software :)

i don't have an extra analysis software. i used opentracks for that. are there good opensource ones compatible with opentracks (i assume the answer is yes). maybe someone can give me a hint which software to choose from.

> And we have no lap feature (based upon reaching the same location multiple times).

would be a very awesome feature :D

> Do you need the statistics during the exercise or afterwards?

both. to keep track of the laps while doing them motivates to try to be faster as the last/best lap. afterwards it is interesting to see how the lap speed increased/decreased over time.

> And which tool do you use to do your analytics?

none, haven't known i would need that, thought thats what opentracks is for but after you ask me this, is see its "opentracks" and not "opentrack-statistics" xD

> would it make more sense to integrate something like this into Fitrackee?

just for reference and the googles https://github.com/SamR1/FitTrackee (on startpage.com "Fitrackee" returns no results at all. looks like startpage is a little bit fussy about the missing "t" xD)

> My argument goes along the line that OprnTracks is running on a mobile only and fancy statistics are not that easy to do there; especially due to the usually limited screen size.

i would partly argue against it. having such features in Fittrackee is of course a very good thing but this wouldn't make them less usable in opentracks. setting up fittrackee requires a server and technical know how.
as i noted above having the lap info while running/... is helpful and motivating. missing such a "basic" feature which quite every simple physical stopwatch has is quite a bummer.
of course i understand that such a feature requires time and money to implement. i just want to say such a feature is quite missing from opentracks and shouldn't be moved to some other app/program with a "need-not-to-be-here"-argument. it would be awesome to have such a feature while running and without the need to be a techie to get them.
